File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        maxArea=0

        while( right > left):
            currentArea = min(height[left],height[right]) * (right-left)
            maxArea = max(currentArea,maxArea)

            if height[right] > height[left]:
                left=left+1
            else:
                right = right -1 

        return maxArea


        

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
let maxVol = 0 

let i = 0

let j = height.length - 1 


while (j>i) {
    let currentVol =  ( j-i )  * Math.min(height[i], height[j] )

    maxVol = Math.max(maxVol,currentVol)
    
    if (height[i] > height[j]){
        j = j-1
    }
    else{
        i = i+1 
    }



}
return maxVol;
};